testing/trino-testing-containers/src/main/java/io/trino/testing/containers/Httpd.java

```diff
@@ -0,0 +1,175 @@
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.log.Logger;
+import org.testcontainers.containers.Network;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+
+public class Httpd
+        extends BaseTestContainer
+{
+    private static final Logger log = Logger.get(Httpd.class);
+
+    public static final String DEFAULT_IMAGE = "httpd:2.4.51";
+    public static final int DEFAULT_LISTEN_PORT = 8888;
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static Builder builderForHttpProxy(int proxyPort)
+    {
+        return new Builder()
+                .withHostName("localhost")
+                .withListenPort(proxyPort)
+                .addLoadModule("mpm_event_module", "modules/mod_mpm_event.so")
+                .addLoadModule("authn_file_module", "modules/mod_authn_file.so")
+                .addLoadModule("authn_core_module", "modules/mod_authn_core.so")
+                .addLoadModule("authz_host_module", "modules/mod_authz_host.so")
+                .addLoadModule("authz_groupfile_module", "modules/mod_authz_groupfile.so")
+                .addLoadModule("authz_user_module", "modules/mod_authz_user.so")
+                .addLoadModule("authz_core_module", "modules/mod_authz_core.so")
+                .addLoadModule("access_compat_module", "modules/mod_access_compat.so")
+                .addLoadModule("auth_basic_module", "modules/mod_auth_basic.so")
+                .addLoadModule("reqtimeout_module", "modules/mod_reqtimeout.so")
+                .addLoadModule("filter_module", "modules/mod_filter.so")
+                .addLoadModule("mime_module", "modules/mod_mime.so")
+                .addLoadModule("log_config_module", "modules/mod_log_config.so")
+                .addLoadModule("env_module", "modules/mod_env.so")
+                .addLoadModule("headers_module", "modules/mod_headers.so")
+                .addLoadModule("setenvif_module", "modules/mod_setenvif.so")
+                .addLoadModule("version_module", "modules/mod_version.so")
+                .addLoadModule("proxy_module", "modules/mod_proxy.so")
+                .addLoadModule("proxy_connect_module", "modules/mod_proxy_connect.so")
+                .addLoadModule("proxy_http_module", "modules/mod_proxy_http.so")
+                .addLoadModule("unixd_module", "modules/mod_unixd.so")
+                .addLoadModule("status_module", "modules/mod_status.so")
+                .addLoadModule("autoindex_module", "modules/mod_autoindex.so")
+                .addLoadModule("dir_module", "modules/mod_dir.so")
+                .addLoadModule("alias_module", "modules/mod_alias.so");
```

Comment:
Sounds like a lot. Are all these actually needed?

Reply:
Yes i've taken it from Product tests where Proxy was enabled.
https://github.com/Praveen2112/trino/blob/f5eb730140fdffe02c33445b92712a10418e5d78/testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-oauth2-http-proxy/httpd.conf
I tried to limit them but failed, seems all are required to enable virtual hosts.